spec/services/post_alerter_spec.rb

```diff
@@ -1194,6 +1194,24 @@ def set_topic_notification_level(user, topic, level_name)
       end
       expect(events).to include(event_name: :before_create_notifications_for_users, params: [[user], post])
     end
+
+    it "does not notify user who edits topic category" do
+      CategoryUser.set_notification_level_for_category(user, CategoryUser.notification_levels[:watching_first_post], category.id)
+      post.update!(last_editor_id: user.id)
+
+      post_alerter = PostAlerter.new
+      post_alerter.notify_first_post_watchers(post, post_alerter.category_watchers(post.topic))
```

Comment:
I'm not quite sure if this is the right spec to add here because we're unit testing the `PostAlerter#notify_first_post_watchers` method in a integration test context. Also, passing `post_alerter.category_watchers(post.topic)` directly to `PostAlerter#notify_first_post_watchers` is an internal implementation concern. If the goal is to unit test the `PostAlerter#notify_first_post_watchers` method, the notification level of the user shouldn't matter because the method takes in the `post` and `user_ids` arguments which has no context attached to them. Instead of calling `PostAlerter#notify_first_post_watcher`, I think we should be testing `PostAlerter.post_edited` instead with the right notification level set for the user.

Reply:
I agree. I added some specs to the jobs that are using PostAlerter and are scheduled when the category or tags are edited.